internal/refactoring/move_execute.go

```diff
@@ -200,10 +200,15 @@ func buildMoveStatementGraph(stmts []MoveStatement) *dag.AcyclicGraph {
 	for dependerI := range stmts {
 		depender := &stmts[dependerI]
 		for dependeeI := range stmts {
+			if dependerI == dependeeI {
+				// skip comparing the statement to itself
+				continue
+			}
 			dependee := &stmts[dependeeI]
-			dependeeTo := dependee.To
-			dependerFrom := depender.From
-			if dependerFrom.CanChainFrom(dependeeTo) || dependerFrom.NestedWithin(dependeeTo) {
+
+			// chain-able moves are simple, as the start and endpoints can only
+			// match one-way
```

Comment:
I think this comment is now redundant, because you (reasonably) moved it into the `statementDependsOn` function below.